src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -229,10 +243,26 @@ object Directory {
 
     // Hashmap from directory path to Directory object, used below for quick access from path.
     val pathToDirectory = HashMap[Path, Directory]()
-
+    var maxId = maxFileId
     for ((dirPath, files) <- leafDirToChildrenFiles) {
-      val allFiles = ListBuffer[FileInfo]()
-      allFiles.appendAll(files.map(FileInfo(_)))
+      val allFiles = fileNameToIdMap match {
+        case Some(map) =>
+          map.sizeHint(files.length) // add size hint for performance improvement.
```

Comment:
This doesn't need to called multiple times. Could you move this out of the loop?